docs/builders/qemu.mdx

```diff
@@ -205,6 +207,77 @@ necessary for this build to succeed and can be found further down the page.
 
 @include 'packer-plugin-sdk/communicator/Config-not-required.mdx'
 
+### SSH key pair automation
+
+The QEMU builder can inject the current SSH key pair's public key into
+the template using the `SSHPublicKey` template engine. This is the SSH public
+key as a line in OpenSSH authorized_keys format.
+
+When a private key is provided using `ssh_private_key_file`, the key's
+corresponding public key can be accessed using the above engine.
+
+@include 'packer-plugin-sdk/communicator/SSH-Private-Key-File-not-required.mdx'
+
+If `ssh_password` and `ssh_private_key_file` are not specified, Packer will
+automatically generate en ephemeral key pair. The key pair's public key can
+be accessed using the template engine.
+
+For example, the public key can be provided in the boot command as a URL
+encoded string by appending `| urlquery` to the variable:
+
+In JSON:
+
+```json
+"boot_command": [
+  "<up><wait><tab> text ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ks.cfg PACKER_USER={{ user `username` }} PACKER_AUTHORIZED_KEY={{ .SSHPublicKey | urlquery }}<enter>"
```

Comment:
Out of curiosity, why the `urlquery` for the public key? If this is raw base64 encoded data, there shouldn't be anything problematic here, is there?
Or maybe spaces, is that the reason why?
If so, maybe escaping the payload would be enough?

Don't get me wrong it's an example so it's fine if it's not perfect/robust in every case, the current implementation should work also.

Reply:
Probably the spaces, this was just copied from the [virtualbox-iso documentation](https://developer.hashicorp.com/packer/integrations/hashicorp/virtualbox/latest/components/builder/iso#ssh-key-pair-automation)

I will run make generate and commit the files